### DISCUSSION TITLE
How to use xrepo emscripten for pakckage test

### DISCUSSION THREAD

**Discussion by star-hengxing** [xmake-io/xmake#5575] [Q&A]
```console
checking for architecture ... wasm32
checking for emsdk directory ... no
checking for emcc.bat ... no
checking for emcc ... no
error: target(test): toolchain not found!
error: execv(xmake f -c -v --plat=wasm --vs_runtime=MD) failed(-1)
```

**Comment by waruqi** [comment]
? 贴下 xmake.lua 

https://github.com/tboox/tbox/blob/553c32406a09e3e0a0a8e8dfed741b4888ba07ca/xmake.lua#L33-L36

**Reply by star-hengxing** [reply]
是在包仓库调用测试脚本，感觉是生成的 xmake.lua 没支持？

**Reply by waruqi** [reply]
没懂，包仓库的测试脚本里不是原本就有 wasm 么

https://github.com/xmake-io/xmake-repo/blob/80e24589ed1b263894fc6d8af9b950f7787d7290/.github/workflows/wasm_ubuntu.yml#L39

**Reply by star-hengxing** [reply]
ci 这个应该是安装时设置了 emsdk 环境变量

而包测试 `xmake l scripts/test.lua --shallow -vD` 生成的 xmake 默认工程没这几行的，所以找不到
https://github.com/tboox/tbox/blob/553c32406a09e3e0a0a8e8dfed741b4888ba07ca/xmake.lua#L33-L36

**Reply by waruqi** [reply]
你是要把外置挪到内置test？ 为啥？方便本地测试？

**Reply by star-hengxing** [reply]
> 方便本地测试？

对，不过内置支持好像有点麻烦，还是自己下个 emsdk 吧
